install.sh

```diff
@@ -1,23 +1,35 @@
 #!/bin/bash
 
-# This script will create links for the dotfiles.
-
-##########
-#  nvim  #
-##########
-mkdir -p "$HOME/.config/nvim"
-mkdir -p "$HOME/.config/nvim/undo"
-ln -sf "$HOME/dotfiles/nvim/init.vim" "$HOME/.config/nvim/init.vim"
+# TODO(collins): add ~/.config/nvim/init.vim to dotfiles/ in this git repo
+# TODO(collins): add ~/.config/X11 to dotfiles/ in this git repo
+# TODO(collins): add ~/.vim to dotfiles/ in this git repo
+# TODO(collins): add ~/.vimrc to dotfiles/ in this git repo
 
+# This script will create links for the dotfiles.
 
-##############
-# Xresources #
-##############
-ln -sfn "$HOME/dotfiles/X11" "$HOME/.config"
-
+# Step 1: ensure all necessary directories exist before linking files.
+for dir in $(find dotfiles/ -mindepth 1 -type d); do
+	target="$HOME/${dir#dotfiles/}"
+	echo "INFO: Ensuring directory exists: $target"
+	mkdir -p $target
+done
 
-########
-#  vim #
-########
-ln -s "$HOME/dotfiles/vim/.vim" "$HOME/.vim"
-ln -s "$HOME/dotfiles/vim/.vimrc" "$HOME/.vimrc"
+# Step 2: create links for all dotfiles
+for file in $(find dotfiles/ -mindepth 1 -type f); do
```

Comment:
This `for` loop is creating links to all the configuration files in the local `dotfiles/` directory in this repository. It is up to you to decide which files are portable/relevant, whichever you decide should be added to this repo so that the links will be created when you run `install.sh`.

Reply:
I run the script then I perform the `TODO's` or copy/move the configuration file to the local `dotfiles/` directory then I run the script which comes first between the 2

Reply:
First delete the example files I added in `dotfiles/`, then copy/move the dotfiles you want to link into `dotfiles/`, then run `install.sh`.